### PR TITLE
Implement OAuth support

### LINKED ISSUE #2: OAuth support
I found this repo by reading this issue: https://github.com/Ecwid/ecwid-mailchimp/issues/31

At last I found a Java wrapper for the Mailchimp v3.0 API. Thank you for doing this.

I was wondering if the wrapper supports Mailchimp OAuth. I saw that you are using the API Key for the requests, but is it possible to replace it with the OAuth access token, that I receive when a user authenticates my app?

Thanks.

### PR DESCRIPTION
**Pull request by ivanst-stoyanov**
I want to resolve #2 and add an OAuth support for the wrapper.

### CODE REVIEW

File: src/main/java/com/mailchimp/MailChimpFactory.java [hunk 1]
```diff
@@ -12,13 +13,23 @@ static String mailChimpApi (String i)
         return "https://"+i+".api.mailchimp.com";
     }
 
-    public static MailChimpClient create (String apiKey, String apiBase)
+    public static MailChimpClient createWithBasicAuth (String apiKey, String apiBase)
     {
         MailChimpClient mailChimp = Feign.builder()
```
Comment (micheal-swiggs) [inline src/main/java/com/mailchimp/MailChimpFactory.java:18]: Thanks for this! Just one thing please keep indentation whitespace at 4 spaces.

Comment (ivanst-stoyanov) [inline src/main/java/com/mailchimp/MailChimpFactory.java:18]: No problem. I am also adding additional functionality like creating lists and batches. I will make separate pull requests for them.

**Comment by micheal-swiggs** [comment]
Is it possible to remove the commit 0cc38414ae ? As you have already created a pull request with this commit. I'm seeing lots of duplicated code on both pull requests.

**Comment by ivanst-stoyanov** [comment]
Sorry about that. I didn't see that it was in the same branch.

**Comment by micheal-swiggs** [comment]
Thankyou! I have merged your pull request into master f0db041 .

**Pull request closed without merging**